File: heavy-metal-bake-sale/typescript/src/Exceptions.ts
export class OutOfStockException extends Error {
  constructor(productName: string) {
    super(`${productName} is out of stock`);
    this.name = 'OutOfStockException';
  }
}

export class InsufficientPaymentException extends Error {
  constructor() {
    super('Not enough money.');
    this.name = 'InsufficientPaymentException';
  }
}

export class UnknownPurchaseCodeException extends Error {
  constructor(code: string) {
    super(`Unknown purchase code: ${code}`);
    this.name = 'UnknownPurchaseCodeException';
  }
}

File: heavy-metal-bake-sale/python/src/heavy_metal_bake_sale/exceptions.py
class OutOfStockException(Exception):
    def __init__(self, product_name: str) -> None:
        super().__init__(f"{product_name} is out of stock")


class InsufficientPaymentException(Exception):
    def __init__(self) -> None:
        super().__init__("Not enough money.")


class UnknownPurchaseCodeException(Exception):
    def __init__(self, code: str) -> None:
        super().__init__(f"Unknown purchase code: {code}")
